### DISCUSSION TITLE
Explanation of different Firmware versions?

### DISCUSSION THREAD

**Discussion by Manos1966** [hoylabs/OpenDTU-OnBattery#292] [Q&A]
I do not seem to find the explanations for the different firmware versions we have since 21.06.2023

The xxx.factory.bin ones can be used to flash the boards, that I understood.

What is the difference between the other ones, especially between the opendtu-generic.bin and opendtu-generic_esp32.bin?
opendtu-generic.bin
opendtu-generic_esp32.bin
opendtu-generic_esp32s3.bin
opendtu-generic_esp32s3_usb.bin

**Comment by helgeerbe** [comment]
Check the wiki https://github.com/tbnobody/OpenDTU/wiki/Binary-Installation

**Reply by Manos1966** [reply]
Aah! so there it was! 

OK, if I understand correctly, I should **not** be using the opendtu-generic.bin (depraced).
Instead, I should use the opendtu-generic_esp32.bin 
But, what does the _Device Profile mandatory_ Yes / No mean? (main difference between the two versions)

![Different Types of Binaries](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/129660360/380c449c-7613-4239-8c82-3876078a4198)

**Comment by schlimmchen** [comment]
A device profile is defined in the `pin_mapping.json`. There are examples in the `docs/DeviceProfiles` directory.

**Reply by Manos1966** [reply]
> A device profile is defined in the pin_mapping.json. 

OK, the pinout also. Thanks @schlimmchen 
Is this the only difference between opendtu-generic.bin and opendtu-generic_esp32.bin ?
I did the whole cable setup (NR24+, Ve.Direct, CANbus) on my ESP32 38pin systems according to the default pins so, I could use the opendtu-generic.bin without worries.

And, when I use a different ESP32 (like the D1 mini on Helge's PCB), I better use the opendtu-generic_esp32.bin

Did I understand the concept correctly?

**Reply by schlimmchen** [reply]
I guess the generic versions with a built-in device profile will eventually be phased out. Always use the generic_esp32 (unless your ESP32 is incompatible, of course, but if that's the case you know about it because you asked for a special part) and configure the device profile as you need it.

**Comment by iukbox** [comment]
Also wenn ich die opendtu-generic_esp32.bin an adresse 0 flashe, dann läuft das flashen durch, danach passiert aber nichts. Kein AP nada.

Serielle Ausgabe:
rst:0x10 (RTCWDT_RTC_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
flash read err, 1000
ets_main.c 371
ets Jun  8 2016 00:22:57

Und den Unterschied zwischen der opendtu-generic_esp32.bin und der opendtu-generic_esp32.factory.bin erschliest sich mir auch nicht.

**Comment by schlimmchen** [comment]
Ich mags dir jetzt nicht raussuchen aus der Doku, aber ich weiß, dass es erklärt ist. Du kannst die non-factory Binaries nicht an Adresse 0 des Flash schreiben. Diese Dateien sind die nackte Firmware, ohne Partitionsschema, ohne Bootloader, etc. Wenn du den Flash neu beschreibst, also ab Adresse 0, dann brauchst du die Datei mit Factory im Namen. Die enthält auch eine Partitionstabelle, einen Bootlader, etc. Und das wäre dann auch der Unterschied zwischen den Dateien.

Die non-factory kannst du, nachdem einmal eine factory-Datei geschrieben wurde, auch in den Flash schreiben, wenn du denn genau weißt, wohin. Das ist kein vorgesehener Anwendungsfall. Die non-factory Dateien sind dafür gemacht, über das Webinterface hochgeladen zu werden.

**Comment by Manos1966** [comment]
Solution HowTo Flash using individual files
![HowTo Flash Solution using four files](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/129660360/0aea4d9a-ea49-4f53-99b1-26da2f799c08)

Solution HowTo Flash using single file "FACTORY"
![HowTo Flash Solution using single file FACTORY](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/129660360/3b2fc400-ee7d-4ffb-9d14-78c37c6ce3d8)
